### PR TITLE
Add custom-carbon-log-appender

### PR DESCRIPTION
**Pull request by vihanga-liyanage**

### CODE REVIEW

File: sample-customizations/custom-carbon-log-appender/LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (Kavindu-Dodan) [inline sample-customizations/custom-carbon-log-appender/LICENSE]: @vihanga-liyanage I do not think we need to include this license again. We already  have one in root level of sample-is

File: sample-customizations/custom-carbon-log-appender/pom.xml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.apache.log4j.wso2</groupId>
+            <artifactId>log4j</artifactId>
+            <version>${orbit.version.log4j}</version>
+        </dependency>
+    </dependencies>
+
+    <properties>
+        <orbit.version.log4j>1.2.17.wso2v1</orbit.version.log4j>
```
Comment (Kavindu-Dodan) [inline sample-customizations/custom-carbon-log-appender/pom.xml]: @vihanga-liyanage Let's not mix up how these versions are defined. Either use properties or directly define with dependency

File: sample-customizations/custom-carbon-log-appender/src/main/java/org/wso2/carbon/custom/utils/logging/appenders/CarbonDailyRollingSanitizedFileAppender.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+        int tenantId = AccessController.doPrivileged(new PrivilegedAction<Integer>() {
+            public Integer run() {
+                return CarbonContext.getThreadLocalCarbonContext().getTenantId();
+            }
+        });
+
+        String serviceName = CarbonContext.getThreadLocalCarbonContext().getApplicationName();
+
+        final LoggingEvent newLoggingEvent = new LoggingEvent(loggingEvent.getFQNOfLoggerClass(), loggingEvent.getLogger(),
```
Comment (Kavindu-Dodan) [inline sample-customizations/custom-carbon-log-appender/src/main/java/org/wso2/carbon/custom/utils/logging/appenders/CarbonDailyRollingSanitizedFileAppender.java]: shall we improve formatting of this block ?

File: sample-customizations/custom-carbon-log-appender/pom.xml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    </properties>
+
+    <!--org.apache.log4j; version="1.2.17"<org.wso2.carbon.logging_4.4.3 [319]>-->
```
Comment (madurangasiriwardena) [inline sample-customizations/custom-carbon-log-appender/pom.xml]: Remove commented code.

File: sample-customizations/custom-carbon-log-appender/src/main/java/org/wso2/carbon/custom/utils/logging/appenders/CarbonDailyRollingSanitizedFileAppender.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
+/*
+*  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (madurangasiriwardena) [inline sample-customizations/custom-carbon-log-appender/src/main/java/org/wso2/carbon/custom/utils/logging/appenders/CarbonDailyRollingSanitizedFileAppender.java]: Shouldn't this be 2019?

**Comment by madurangasiriwardena** [comment]
@vihanga-liyanage can you please attend the comments added?

**Comment by vihanga-liyanage** [comment]
@Kavindu-Dodan, @madurangasiriwardena Comments were addressed. Sorry about the delay.

**Pull request merged**